File: leetcode/lc-us/cpp/2422.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int minimumOperations(vector<int> &nums) {
        int n = (int) nums.size();
        int i = 0, j = n - 1;
        int res = 0;
        long long l = nums[0], r = nums[n - 1];
        while (i < j) {
            if (l != r) {
                if (l < r) {
                    i++;
                    l += nums[i];
                } else {
                    j--;
                    r += nums[j];
                }
                res++;
            } else {
                i++, j--;
                l = nums[i], r = nums[j];
            }
        }
        return res;
    }
};

File: leetcode/lc-us/py/2422.py
from typing import List


class Solution:
    def minimumOperations(self, nums: List[int]) -> int:
        cnt = 0
        i, j = 0, len(nums) - 1
        while i < j:
            while nums[i] < nums[j]:
                nums[i + 1] += nums[i]
                i += 1
                cnt += 1
            while nums[j] < nums[i]:
                nums[j - 1] += nums[j]
                j -= 1
                cnt += 1
            if nums[i] == nums[j]:
                i += 1
                j -= 1
            else:
                continue
        return cnt
